internal/test/unstructured.go

```diff
@@ -0,0 +1,172 @@
+// Package test provides utilities for testing unstructured Kubernetes objects.
+//
+// The primary functionality is JSONPath-like field access for unstructured.Unstructured objects,
+// making test assertions more readable and maintainable.
+package test
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// FieldString retrieves a string field from an unstructured object using JSONPath-like notation.
+// Examples:
+//   - "spec.runStrategy"
+//   - "spec.template.spec.volumes[0].containerDisk.image"
+//   - "spec.dataVolumeTemplates[0].spec.sourceRef.kind"
+func FieldString(obj *unstructured.Unstructured, path string) string {
+	if obj == nil {
+		return ""
+	}
+	value, _ := Field(obj.Object, path)
+	if str, ok := value.(string); ok {
+		return str
+	}
+	return ""
+}
+
+// FieldExists checks if a field exists at the given JSONPath-like path.
+func FieldExists(obj *unstructured.Unstructured, path string) bool {
+	if obj == nil {
+		return false
+	}
+	_, found := Field(obj.Object, path)
+	return found
+}
+
+// FieldInt retrieves an integer field from an unstructured object using JSONPath-like notation.
+// Returns 0 if the field is not found or is not an integer type (int, int64, int32).
```

Comment:
@manusa I'm not 100% sure this behaviour is ideal - there could be scenarios where I'm testing and expecting to receive a value of 0 - for example status.replicas. How would we be able to distinguish between the value actually being 0 vs. not being set/not being an int? 

Or do you think that's fine because this is for tests? In that case I'm fine with it but let's add a note in the docstring about ensuring that you don't assert expecting an actual value to be 0